### PR TITLE
Debug print

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: riscv/tests/riscv_data/sum.rs [hunk 1]
```diff
@@ -12,7 +14,82 @@ pub extern "C" fn main() -> ! {
...
+
+    pub fn show<'a>(buffer: &'a mut [u8], args: fmt::Arguments) -> Result<&'a str, fmt::Error> {
+        let mut w = WriteTo::new(buffer);
+        fmt::write(&mut w, args)?;
+        w.as_str().ok_or(fmt::Error)
+    }
+}
+
+#[panic_handler]
+fn panic(info: &PanicInfo) -> ! {
```
Comment (leonardoalt) [inline riscv/tests/riscv_data/sum.rs:83]: Should we put all this stuff in a separate lib?

**Comment by leonardoalt** [comment]
tests failing

**Comment by leonardoalt** [comment]
tests failing

**Comment by leonardoalt** [comment]
tests failing

File: compiler/src/lib.rs [hunk 2]
```diff
@@ -92,16 +92,24 @@ pub fn compile_asm_string(
...
                 let index = items[1].parse::<usize>().unwrap();
                 let value = inputs.get(index).cloned();
                 if let Some(value) = value {
                     log::trace!("Input query: Index {index} -> {value}");
                 }
                 value
             }
+            "\"print\"" => {
+                log::info!("Print: {}", items[1..].join(", "));
+                Some(0.into())
```
Comment (leonardoalt) [inline compiler/src/lib.rs:107]: bit of a hack innit :p

**Review by leonardoalt — APPROVED** [review]

File: compiler/src/lib.rs (hunk 2 above)
Comment (chriseth) [reply compiler/src/lib.rs:107]: Sorry, what do you mean? We can also remove it again, I'm only using `print_ch`.

File: riscv/runtime/src/print.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+use core::fmt;
+
+// #[macro_export]
+// macro_rules! print {
+//     ($($arg:tt)+) => (print_args(format_args!( $($arg)+)))
+// }
+
+// TODO turn this into a macro
+pub fn print(args: fmt::Arguments) {
+    let mut buf = [0u8; 1024];
```
Comment (chriseth) [inline riscv/runtime/src/print.rs:11]: Let me actually change this: We can use "potentially uninitialized" - then it should be much faster.

File: compiler/src/lib.rs (hunk 2 above)
Comment (leonardoalt) [reply compiler/src/lib.rs:107]: I mean returning Some(0)

**Pull request merged**